src/test/token/ERC20Gauges.t.sol

```diff
@@ -0,0 +1,512 @@
+        require(token.getUserWeight(address(1)) == 3e18);
+        require(token.getGaugeWeight(gauge2) == 5e18);
+        require(token.totalWeight() == 10e18);
+    }   
+
+    function testFailIncrementOnDeprecated() public {
+        token.setMaxGauges(2);
+        token.addGauge(gauge1);
+        token.removeGauge(gauge1);
+
+        token.incrementGauge(gauge1, 1e18);
+    }
+
+    function testFailIncrementOverWeight() public {
+        token.setMaxGauges(2);
+        token.addGauge(gauge1);
+        token.addGauge(gauge2);
+
+        require(token.incrementGauge(gauge1, 50e18) == 50e18);
+        token.incrementGauge(gauge2, 51e18);
+    }
+
+    function testIncrementGauges() public {
+        token.setMaxGauges(2);
+        token.addGauge(gauge1);
+        token.addGauge(gauge2);
+
+        token.incrementGauge(gauge1, 1e18);
+
+        address[] memory gaugeList = new address[](2);
+        uint[] memory weights = new uint[](2);
+        gaugeList[0] = gauge2;
+        gaugeList[1] = gauge1;
+        weights[0] = 2e18;
+        weights[1] = 4e18;
+
+        require(token.incrementGauges(gaugeList, weights) == 7e18);
+
+        require(token.getUserGaugeWeight(address(this), gauge2) == 2e18);
+        require(token.getGaugeWeight(gauge2) == 2e18);
+        require(token.getUserGaugeWeight(address(this), gauge1) == 5e18);
+        require(token.getUserWeight(address(this)) == 7e18);
+        require(token.getGaugeWeight(gauge1) == 5e18);
+        require(token.totalWeight() == 7e18);
+    }
+
+    function testFailIncrementGaugesDeprecated() public {
+        token.setMaxGauges(2);
+        token.addGauge(gauge1);
+        token.addGauge(gauge2);
+        token.removeGauge(gauge2);
+
+        address[] memory gaugeList = new address[](2);
+        uint[] memory weights = new uint[](2);
+        gaugeList[0] = gauge2;
+        gaugeList[1] = gauge1;
+        weights[0] = 2e18;
+        weights[1] = 4e18;
+
+        token.incrementGauges(gaugeList, weights);    
+    }
+
+    function testFailIncrementGaugesOver() public {
+        token.setMaxGauges(2);
+        token.addGauge(gauge1);
+        token.addGauge(gauge2);
+
+        address[] memory gaugeList = new address[](2);
+        uint[] memory weights = new uint[](2);
+        gaugeList[0] = gauge2;
+        gaugeList[1] = gauge1;
+        weights[0] = 50e18;
+        weights[1] = 51e18;
+
+        token.incrementGauges(gaugeList, weights);    
+    }
+
+    function testFailIncrementGaugesSizeMismatch() public {
+        token.setMaxGauges(2);
+        token.addGauge(gauge1);
+        token.addGauge(gauge2);
+        token.removeGauge(gauge2);
+
+        address[] memory gaugeList = new address[](2);
+        uint[] memory weights = new uint[](3);
+        gaugeList[0] = gauge2;
+        gaugeList[1] = gauge1;
+        weights[0] = 1e18;
+        weights[1] = 2e18;
+
+        token.incrementGauges(gaugeList, weights);    
+    }
+
+    function testDecrement() public {
```

Comment:
For some of these more involved tests, I'd perhaps add a /// @notice natspec comment to explain what they're testing specifically